File: hackerrank/circle_city.py
#!/bin/python
import math

t = int(input())
for _ in range(t):
    r2, k = (int(x) for x in raw_input().split())

    count = 0
    r = int(math.sqrt(r2))
    if r2 - r*r == 0:
        count = 4
    for a in range(1, r+1):
        a2 = a*a
        b = int(math.sqrt(r2-a2))
        if r2 - a2 - b*b == 0:
            if a < b:
                count += 8
            elif a == b:
                count += 4
    if k < count:
        print 'impossible'
    else:
        print 'possible'

File: hackerrank/circle_city.cpp
#include <cstdlib>
#include <cstdio>
#include <cmath>
#include <iostream>

using std::cout;
using std::cin;
using std::endl;

void is_possible(long r2, long k){
    long count = 0, a, b, a2, b2;
    long r = sqrt(r2);
    if(r2 - r == 0)
        count += 4;
    for(a=1; a<r; a++){
        a2 = a*a;
        for(b=1; b<a; b++){
            b2 = b*b;
            if((r2 - a2 - b2) == 0)
                count += 8;
        }
    }
    if(k<count)
        printf("impossible\n");
    else
        printf("possible\n");
}

int main(int argc, char ** argv){
    int t;
    long r2, k;
    cin >> t;
    for(int idx=0; idx<t; idx++){
        cin >> r2 >> k;
        is_possible(r2, k);
    }
}
